pre_award/assess/assessments/routes.py

```diff
@@ -1311,25 +1311,30 @@ def request_changes(application_id, sub_criteria_id, theme_id):
     assessment_status = determine_assessment_status(sub_criteria.workflow_status, state.is_qa_complete)
     theme_answers_response = get_sub_criteria_theme_answers_all(application_id, theme_id)
 
-    form = RequestChangesForm(
-        question_choices=[(question["field_id"], question["question"]) for question in theme_answers_response]
-    )
+    question_choices = [
+        (question["field_id"], question["question"], question.get("answer", "")) for question in theme_answers_response
+    ]
 
+    form = build_request_changes_form(question_choices)()
+    justification_data = {field_id: getattr(form, f"reason_{field_id}").data for field_id, _, _ in question_choices}
```

Comment:
You only really need the `justification_data` inside the POST if, please move it inside.

Also.. I would try to do this field_id to reason_field_id mapping on the form data submitted, not all the fields. That way you will always have have a non-None reason textareas. Saving you an if.